pkg/vpa/service.go

```diff
@@ -149,6 +149,44 @@ func (c *Service) CreateTortoiseUpdaterVPA(ctx context.Context, tortoise *autosc
 	return vpa, tortoise, nil
 }
 
+func (c *Service) UpdateVPACRP(ctx context.Context, tortoise *autoscalingv1beta3.Tortoise) (*v1.VerticalPodAutoscaler, error) {
+	retVPA := &v1.VerticalPodAutoscaler{}
+
+	// we only want to record metric once in every reconcile loop.
+	//metricsRecorded := false
+	updateFn := func() error {
+		vpa, err := c.GetTortoiseUpdaterVPA(ctx, tortoise)
+		if err != nil {
+			return fmt.Errorf("get tortoise VPA: %w", err)
+		}
+		crp := make([]v1.ContainerResourcePolicy, 0, len(tortoise.Spec.ResourcePolicy))
+		for _, c := range tortoise.Spec.ResourcePolicy {
+			crp = append(crp, v1.ContainerResourcePolicy{
+				ContainerName: c.ContainerName,
+				MinAllowed:    c.MinAllocatedResources,
+			})
+		}
+		vpa.Spec.ResourcePolicy.ContainerPolicies = crp
```

Comment:
We have to care the possibility where vpa.Spec.ResourcePolicy is nil in the first place.
Given we don't use any other field in ResourcePolicy, we can change here to:

```suggestion
		vpa.Spec.ResourcePolicy = &v1.PodResourcePolicy{ContainerPolicies: crp}
```